pkg/executor/openstack.go

```diff
@@ -0,0 +1,561 @@
+// Copyright (c) 2018 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+Openstack executor under the hood is using these are two components:
+- executor:
+	- talks to Openstack cluster to create new instance,
+	- starts watcher (by calling Execute),
+	- runs in main goroutine,
+	- gives back control to user by returning newly created taskHandle
+- watcher:
+	- responsible for monitoring state of Instance, passing to information to taskHandle
+	- also in case of failure or part of cleaning up or when asked directry by taskHandle - delete instance
+*/
```

Comment:
Believe it or not but there are strict requirements for [documenting Go code](https://blog.golang.org/godoc-documenting-go-code) ;)